static/app/components/compactSelect/index.tsx

```diff
@@ -0,0 +1,128 @@
+import {useMemo} from 'react';
+import {Item, Section} from '@react-stately/collections';
+
+import domId from 'sentry/utils/domId';
+
+import {Control, ControlProps} from './control';
+import {ListBox, MultipleListBoxProps, SingleListBoxProps} from './listBox';
+import type {
+  SelectOption,
+  SelectOptionOrSection,
+  SelectOptionOrSectionWithKey,
+  SelectSection,
+} from './types';
+
+export {SelectOption, SelectOptionOrSection, SelectSection};
+
+interface BaseSelectProps<Value extends React.Key> extends ControlProps {
+  options: SelectOptionOrSection<Value>[];
+  /**
+   * Same as `disabled` in ControlProps. Allowed only for backward compatibility.
+   * Will be removed soon.
+   * @deprecated
+   */
+  isDisabled?: boolean;
+}
+
+export interface SingleSelectProps<Value extends React.Key>
+  extends BaseSelectProps<Value>,
+    Omit<SingleListBoxProps<Value>, 'children' | 'items' | 'compositeIndex' | 'label'> {}
+
+export interface MultipleSelectProps<Value extends React.Key>
+  extends BaseSelectProps<Value>,
+    Omit<
+      MultipleListBoxProps<Value>,
+      'children' | 'items' | 'compositeIndex' | 'label'
+    > {}
+
+export type SelectProps<Value extends React.Key> =
+  | SingleSelectProps<Value>
+  | MultipleSelectProps<Value>;
+
+// A series of TS function overloads to properly parse prop types across 2 dimensions:
+// option value types (number vs string), and selection mode (singular vs multiple)
+function CompactSelect<Value extends number>(props: SelectProps<Value>): JSX.Element;
+function CompactSelect<Value extends string>(props: SelectProps<Value>): JSX.Element;
+function CompactSelect<Value extends React.Key>(props: SelectProps<Value>): JSX.Element;
+
+/**
+ * Flexible select component with a customizable trigger button
+ */
+function CompactSelect<Value extends React.Key>({
+  // List box props
+  options,
+  value,
+  defaultValue,
+  onChange,
+  multiple,
+  disallowEmptySelection,
+  isOptionDisabled,
+
+  // Control props
+  disabled,
+  isDisabled,
+  size = 'md',
+  closeOnSelect,
+  triggerProps,
+  ...controlProps
+}: SingleSelectProps<Value> | MultipleSelectProps<Value>) {
```

Comment:
Can't this be `SelectProps<Value>`

Reply:
Right forgot to switch that out!

Reply:
https://github.com/getsentry/sentry/pull/43449/commits/38f2eaa259fe4b6f645d0e0398c99983c02163d5